### ISSUE TITLE
Pytorch implementation

### ISSUE THREAD

**Issue by max-vasyuk** [felixrosberg/FaceDancer#9]
Thank you for this amazing work! I too am in the process of implementing this approach on PyTorch. Could you please tell me exactly what the problems are? And how does it affect the results of the network?

**Comment by felixrosberg** [comment]
Hello @max-vasyuk ,

Cool!

Well currently I suspect I have screwed the gradients over due to difference between PyTorch and TensorFlow, as I naively implemented it first straight of as I did in TensorFlow, but I am unsure. The PyTorch version seem to not be able to learn to tranfer the id when using a mapping network. I also experimented around having 20% chance of the same face as described in the paper, vs, every 20% batch make source and target the same, which first seemed to fix a few things.

**Comment by joe-zxh** [comment]
Hello @felixrosberg 
FaceDancer is a great work! Could you release the pytorch version of FaceDancer even it has some problems now?
The community (including me :-)) may try to figure out the problem.

Thanks and I am looking forward to hearing from you.

**Comment by joe-zxh** [comment]
I think the problem of "20% chance of the same face" is causing by the cycle loss Lc. Generating the Xc same as Xt will easily generate a Xt' same as Xt, which can lower the Lc loss.

![image](https://github.com/felixrosberg/FaceDancer/assets/29706834/d9c6125e-0d2b-4737-88fc-73f64b288171)

**Comment by joe-zxh** [comment]
@max-vasyuk Hi there, will you release your pytorch version implementaion of FaceDancer? I am not familiar with the tensorflow code.

Thanks and I am looking forward to hearing from you.
